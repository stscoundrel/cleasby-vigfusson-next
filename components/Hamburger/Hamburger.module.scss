.hamburger {
  cursor: pointer;
  width: 24px;
  height: 18px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.part {
  display: block;
  width: 100%;
  height: 2px;
  background-color: white;
  transition: all 0.2s ease-out;
}

.open {
  span {
    &:nth-child(1) {
      position: relative;
      top: 9px;
      transform: rotate(45deg);
    }
    &:nth-child(2) {
      display: none;
    }
    &:nth-child(3) {
      position: relative;
      top: -7px;
      transform: rotate(-45deg);
    }
  }
}