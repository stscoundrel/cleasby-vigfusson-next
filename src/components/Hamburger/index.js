import { useState } from 'react'
import styles from './Hamburger.module.scss'

export default function Hamburger({ action }) {
  const [isOpen, setIsOpen] = useState(false)

  const openMenu = () => {
    setIsOpen(!isOpen)
    action()
  }

  const getHamburgerClass = () => {
    if (isOpen) {
      return `${styles.hamburger} ${styles.open}`
    }

    return styles.hamburger
  }

  return (
    <div role="button" aria-label="Open menu" className={getHamburgerClass()} onClick={() => openMenu()}>
        <span className={styles.part} />
        <span className={styles.part} />
        <span className={styles.part} />
    </div>
  )
}
