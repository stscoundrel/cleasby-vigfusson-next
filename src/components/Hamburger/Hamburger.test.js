import React from 'react'
import renderer from 'react-test-renderer'
import Hamburger from './index'

describe('Hamburger button', () => {
  test('Matches the snapshot', () => {
    const mockHandler = jest.fn()
    const tree = renderer.create(
      <Hamburger action={mockHandler} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Triggers callback after click', async () => {
    const mockHandler = jest.fn()

    const tree = renderer.create(
      <Hamburger action={mockHandler} />,
    )

    // Hamburger is closed at start.
    const closedState = tree.toJSON()

    // Click hamburger.
    await renderer.act(async () => {
      await tree.root.findByType('div').props.onClick()

      // Assert hamburger has open stat & callback was called.
      const openState = tree.toJSON()
      expect(closedState).not.toEqual(openState)
      expect(mockHandler).toHaveBeenCalled()
    })
  })
})
