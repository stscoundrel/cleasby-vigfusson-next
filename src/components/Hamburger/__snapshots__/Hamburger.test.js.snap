// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger button Matches the snapshot 1`] = `
<div
  aria-label="Open menu"
  className="hamburger"
  onClick={[Function]}
  role="button"
>
  <span
    className="part"
  />
  <span
    className="part"
  />
  <span
    className="part"
  />
</div>
`;
